lib/ae_mdw/contract.ex

```diff
@@ -346,6 +348,26 @@ defmodule AeMdw.Contract do
     Enum.group_by(get_events(micro_block), fn {_event_name, %{tx_hash: tx_hash}} -> tx_hash end)
   end
 
+  @spec maybe_resolve_contract_pk(name_pubkey() | contract_pubkey()) :: contract_pubkey()
+  def maybe_resolve_contract_pk(contract_or_name_pk) do
+    contract_or_name_pk
+    |> case do
+      <<217, 52, 92, _rest::binary>> = name_pk ->
+        "contract_pubkey"
+        |> :aec_chain.resolve_namehash(name_pk)
+        |> case do
+          {:ok, {:id, :contract, pubkey}} ->
+            pubkey
+
+          {:error, _} ->
```

Comment:
Not sure why credo didn't catch this but unused variables should contain a reference to what it actually is (`_reason` in this case). But maybe it would be even better to include the reason as part of the error being raised